content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java

```diff
@@ -295,6 +304,50 @@ public boolean entityExists(String id) {
         }
 		return rv;
 	}
+
+	/**
+	 * return the original URL string based on Web Link resource content 
+	 * @param resource
+	 * @param contentType
+	 * @return
+	 */
+	private String getWebLinkResourceUrlString(ContentResource resource, String contentType) {
+		String urlString = null;
+		
+		// for Web Link resources
+		// return the original URL link, instead of access URL
+		if (contentType.equalsIgnoreCase(ResourceProperties.TYPE_URL))
+		{
+			try
+			{
+				byte[] content = resource.getContent();
+				if (content != null && content.length > 0)
+				{
+					try
+					{
+						// An invalid URI format will get caught by the outermost catch block 
+						URI uri = new URI(new String(content, "UTF-8"));
+						urlString = uri.toString();
+					}
+					catch (UnsupportedEncodingException e)
+					{
+						log.info(this + "getWebLinkResourceUrlString unsupportedencodingexception for " + new String(content) + " " + e.getMessage());
```

Comment:
Hi Zhen,

Can you modify the logging to use warn instead of info when you catch an exception?